src/main/java/org/thoughtcrime/securesms/accounts/AccountSelectionListFragment.java

```diff
@@ -168,11 +168,16 @@ private void onDeleteAccount(int accountId) {
     avatar.setAvatar(GlideApp.with(activity), recipient, false);
     nameView.setText(name);
     addrView.setText(contact.getAddr());
-    try {
-      sizeView.setText(Util.getPrettyFileSize(rpc.getAccountFileSize(accountId)));
-    } catch (RpcException e) {
-      e.printStackTrace();
-    }
+    Util.runOnAnyBackgroundThread(() -> {
+      try {
+        final int sizeBytes = rpc.getAccountFileSize(accountId);
+        Util.runOnMain(() -> {
+          sizeView.setText(Util.getPrettyFileSize(sizeBytes));
```

Comment:
if you cancel the dialog in a GBs account that takes several seconds, would be got to know if this causes some background exception because the dialog was already destroyed

Reply:
but my guess is that this should be fine, since the view is created separately from the dialog

Reply:
i think, there is a strong reference to the object as defined in the outer block, if at all, this could irritate the garbage collection, but even that should be fine as when the callback is finally left, the object can be freed

i tried that in practise by adding a `final int sizeBytes = rpc.getAccountFileSize(accountId);Util.sleep(10000);`  and closing all dialogs before the 10 seconds, everything seems fine